libfwupd/fwupd-common.c

```diff
@@ -817,6 +817,34 @@ fwupd_guid_hash_data (const guint8 *data, gsize datasz, FwupdGuidFlags flags)
 	return fwupd_guid_to_string ((const fwupd_guid_t *) &uu_new, flags);
 }
 
+/**
+ * fwupd_device_id_is_valid:
+ * @device_id: string to check, e.g. `d3fae86d95e5d56626129d00e332c4b8dac95442`
+ *
+ * Checks the string is a valid non-partial device ID. It is important to note
+ * that the wildcard ID of `*` is not considered a valid ID in this function and
+ * the client must check for this manually if this should be allowed.
+ *
```

Comment:
Why not have a g_return_val_if_fail in this case to prevent coding errors?

Reply:
Good question, but that doesn't seem right. I see this symbol being used for "is a fully specified ID" so I think it's perfectly fine to call it on `*` and get FALSE.

Reply:
OK.